### PR TITLE
Remove order check for ICA host and controller upgrade callbacks

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #5524

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/27-interchain-accounts/controller/keeper/handshake.go [hunk 1]
```diff
@@ -151,17 +151,10 @@ func (Keeper) OnChanCloseConfirm(
 // - encoding (must be supported)
 //
 // The following may not be changed:
-// - order
```
Comment (DimitrisJim) [inline modules/apps/27-interchain-accounts/controller/keeper/handshake.go:154]: can move this to two lines up under `The following may be changed`

File: modules/apps/27-interchain-accounts/host/keeper/handshake.go [hunk 2]
```diff
@@ -141,17 +141,10 @@ func (Keeper) OnChanCloseConfirm(
 // - encoding (must be supported)
 //
 // The following may not be changed:
-// - order
```
Comment (DimitrisJim) [inline modules/apps/27-interchain-accounts/host/keeper/handshake.go:144]: dittorino

**Comment by DimitrisJim** [comment]
should also slap backport label for 8.1 on this

**Comment by crodriguezvega** [comment]
> should also slap backport label for 8.1 on this

I think we could try to target all the PRs for the feature to a feature branch and then backport just the PR for the feature branch.

**Comment by DimitrisJim** [comment]
> we could try to target all the PRs for the feature to a feature branch

sounds perfect!

**Review by colin-axner — APPROVED** [review]
Love how clean the diffs are! Took 2 seconds to review! :heart:

**Review by crodriguezvega — APPROVED** [review]
Sweet!

**Pull request merged**